File: year2017/day02/python/solution.py
def main():
    print(part_1())
    print(part_2())


def part_1():
    return sum(max(row) - min(row) for row in input)


def part_2():
    result = 0
    for row in input:
        div, num = next(((n, m) for i, n in enumerate(row) for j, m in enumerate(row) if i != j and n % m == 0), (None, None))
        result += div / num if div is not None and num != 0 else 0
    return result


def setup(path):
    global input
    
    with open(path, 'r') as file:
        input = [[int(num) for num in row.split('\t')] for row in file.read().split('\n')]


if __name__ == '__main__':
    setup('../puzzle.txt')
    main()

File: year2017/day02/java/Solution.java
import java.nio.file.*;
import java.util.*;

public class Solution {
    private static List<List<Integer>> SPREADSHEET = new ArrayList<>();

    public static void main(String[] args) { 
        try {
            setup(Files.readString(Paths.get("../puzzle.txt")));
            System.out.println(part1());
            System.out.println(part2());
        } catch(Exception e) {
            e.printStackTrace();
        }
    }
    
	public static int part1() {
        int sum = 0;
        for (List<Integer> row : SPREADSHEET) {
            int max = Integer.MIN_VALUE;
            int min = Integer.MAX_VALUE;
            for (int num : row) {
                max = Math.max(max, num);
                min = Math.min(min, num);
            }
            sum += (max - min);
        }
        return sum;
    }
    
    public static int part2() {
        int sum = 0;
        for (List<Integer> row : SPREADSHEET) {
            int div = 0;
            int num = 0;
            loop:
            for (int i = 0; i < row.size(); i++) {
                for (int j = 0; j < row.size(); j++) {
                    if (i != j && row.get(j) != 0 && row.get(i) % row.get(j) == 0) {
                        div = row.get(i);
                        num = row.get(j);
                        break loop;
                    }
                }
            }
            sum += (div / num);
        }
        return sum;
    }

    public static void setup(String input) {
        for (String row : input.split("\n")) {
            List<Integer> ROW = new ArrayList<>();
            for(String num : row.split("\t")) {
                ROW.add(Integer.parseInt(num));
            }
            SPREADSHEET.add(ROW);
        }
    }
}